linker-interface/shared/src/test/scala/org/scalajs/linker/interface/StandardConfigFingerprintTest.scala

```diff
@@ -0,0 +1,134 @@
+/*
+ * Scala.js (https://www.scala-js.org/)
+ *
+ * Copyright EPFL.
+ *
+ * Licensed under Apache License 2.0
+ * (https://www.apache.org/licenses/LICENSE-2.0).
+ *
+ * See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ */
+
+package org.scalajs.linker.interface
+
+import java.net.URI
+
+import org.junit.Test
+import org.junit.Assert._
+
+import Semantics.RuntimeClassNameMapper
+import Semantics.RuntimeClassNameMapper._
+
+class StandardConfigFingerprintTest {
+
+  lazy val standardConfigs: Stream[StandardConfig] = {
+    def modifyField[Field, Obj](
+      values: Stream[Field],
+      fieldValue: Field,
+      fieldSetter: Field => Obj
+    ): Stream[Obj] = {
+      values.filter(_ != fieldValue).map(fieldSetter)
+    }
+
+    lazy val booleanStream: Stream[Boolean] = Stream(true, false)
+
+    lazy val runtimeClassNameMapperStream: Stream[RuntimeClassNameMapper] =
+      Stream(
+        discardAll(),
+        keepAll(),
+        regexReplace("""\d""".r, "0"),
+        keepAll() andThen discardAll()
+      )
+
+    lazy val checkedBehaviorStream: Stream[CheckedBehavior] = Stream(
+      CheckedBehavior.Compliant,
+      CheckedBehavior.Fatal,
+      CheckedBehavior.Unchecked
+    )
+
+    lazy val semanticsStream: Stream[Semantics] = {
```

Comment:
Couldn't this simply be written as:

```scala
lazy val semanticsStream = for {
  asInstanceOfs <- checkedBehaviorStream
  arrayIndexOutOfBounds <- checkedBehaviorStream
  ...
} yield {
  Semantics.Defaults
    .withAsInstanceOfs(asInstanceOfs)
    .withArrayIndexOutOfBounds(arrayIndexOutOfBounds)
    ...
}
```

Seems simpler to me (and I think actually gives us the full cross product, unlike the current code). Then again, less logic in tests is also good. So maybe just a bunch of example cases would be enough.

This comment applies in general to the tests.

Reply:
I tried the full cross product, but it takes a minute to run, which seemed a little too long. Would it be acceptable to have such a long-running test?

Reply:
No, I don't think so. Especially for such a simple thing.

TBH, I'm not sure it is worth trying to make the tests cover every field. Mainly because we cannot write a more general test for this. What I mean by that is: If we add a new field somewhere and we forget the put it in the fingerprint, we'll very likely also forget to put it in the test. So from this POV, exhaustive field testing seems overkill anyways.

What we should test is that the `FingerprintBuilder` works properly (and is wired somewhat reasonably). But that we can test with 2-3 examples. This might be better than complicated logic.

Since it seems that we are sharing our internal documentation, this explains why:
https://testing.googleblog.com/2014/07/testing-on-toilet-dont-put-logic-in.html

Reply:
Thank you, that was a fun read :) I've simplified the `StandardConfig` fingerprint tests to just run a few examples in 27a213e1c7ee420d3bd49913d45a2bbef425cdd3, and I have also added a few tests directly for the `FingerprintBuilder` in d0ad254ba744ba53f654e16763461483aed53822.